src/swap/SwapScreen.tsx

```diff
@@ -762,23 +785,22 @@ export function SwapScreen({ route }: Props) {
           showLoading={confirmSwapIsLoading}
         />
       </ScrollView>
-      <TokenBottomSheet
-        forwardedRef={tokenBottomSheetRef}
-        snapPoints={['90%']}
-        origin={TokenPickerOrigin.Swap}
-        onTokenSelected={handleSelectToken}
-        searchEnabled={true}
-        tokens={selectingField == Field.FROM ? swappableFromTokens : swappableToTokens}
-        title={
-          selectingField == Field.FROM
-            ? t('swapScreen.swapFromTokenSelection')
-            : t('swapScreen.swapToTokenSelection')
-        }
-        TokenOptionComponent={TokenBalanceItemOption}
-        showPriceUsdUnavailableWarning={true}
-        filterChips={filterChips}
-        areSwapTokensShuffled={areSwapTokensShuffled}
-      />
+      {tokenBottomSheetsConfig.map(({ fieldType, tokens, title, filterChips, origin }) => (
+        <TokenBottomSheet
+          key={`TokenBottomSheet/${fieldType}`}
+          forwardedRef={tokenBottomSheetRefs[fieldType]}
+          tokens={tokens}
+          title={title}
+          filterChips={filterChips}
+          origin={origin}
+          snapPoints={['90%']}
+          onTokenSelected={handleSelectToken}
+          searchEnabled={true}
+          TokenOptionComponent={TokenBalanceItemOption}
+          showPriceUsdUnavailableWarning={true}
+          areSwapTokensShuffled={areSwapTokensShuffled}
+        />
+      ))}
```

Comment:
From what I see we need to mount 2 bottom sheets because we want to keep the selected filter chips state separate whether it's "from" vs "to" , right?

What happens when different networks are selected? and the user switches between the from and to? does it feel natural?
I guess I should play with it myself :D

Reply:
yes - we want the "from" and "to" sheets to behave independently, and for the user to pick up where they left off if they go back to either sheet.

interesting question about whether switching networks feels natural. i'm not sure what my personal preference is, but i think it does not feel so clunky  / unintuitive now that the filters are so easily accessible and you can see at a glance what the current selection is. i'm also happy to merge as is and update it depending on feedback once we've been able to test it on a physical device in the nightly build